Assets/Scripts/EnemyAI.cs

```diff
@@ -15,16 +14,19 @@ public class EnemyAI : MonoBehaviour
     [SerializeField] float detectRange = 5;
     [SerializeField] float hearRange = 1.5f;
     float _attackRange;
-    [SerializeField] GameObject bullet;
-    Bullet _bullet;
-    [SerializeField] float shootDelay = 1;
-    bool _canShoot = true;
+    [SerializeField] GameObject projectile;
+    [SerializeField] float attackDelay = 1;
+    bool _canAttack = true;
+    Health _playerHealth;
+    LayerMask _playerLayer;
 
```

Comment:
following Single Responsibility Principle, I think it would be beneficial to separate enemy movement from enemy attacking as this file is already getting really long

Reply:
Yes this file is an absolute mess already. I will try to separate patrolling, chasing, combat and other behaviors in different modular scripts. It will probably require re-implementing the entire system but it's not very good as it is anyway.